x-pack/qa/repository-old-versions/src/test/java/org/elasticsearch/oldrepos/OldRepositoryAccessIT.java

```diff
@@ -130,10 +162,98 @@ public void testOldRepoAccess() throws IOException {
                 assertEquals(0, snapshotStatus.getShardsStats().getFailedShards());
                 assertThat(snapshotStatus.getStats().getTotalSize(), greaterThan(0L));
                 assertThat(snapshotStatus.getStats().getTotalFileCount(), greaterThan(0));
+
+                // so far only added basic infrastructure for reading 6.0.0+ indices
+                if (Build.CURRENT.isSnapshot() && oldVersion.onOrAfter(Version.fromString("6.0.0"))) {
+                    // restore / mount and check whether searches work
+                    restoreMountAndVerify(numDocs, expectedIds, client, numberOfShards);
+
+                    // close indices
+                    assertTrue(
+                        client.indices().close(new CloseIndexRequest("restored_test"), RequestOptions.DEFAULT).isShardsAcknowledged()
+                    );
+                    assertTrue(
+                        client.indices().close(new CloseIndexRequest("mounted_test"), RequestOptions.DEFAULT).isShardsAcknowledged()
+                    );
+
+                    // restore / mount again
+                    restoreMountAndVerify(numDocs, expectedIds, client, numberOfShards);
+                }
             } finally {
                 oldEs.performRequest(new Request("DELETE", "/test"));
             }
         }
     }
 
+    private static String sourceForDoc(int i) {
+        return "{\"test\":\"test" + i + "\",\"val\":" + i + "}";
+    }
+
+    @SuppressWarnings("removal")
+    private void restoreMountAndVerify(int numDocs, Set<String> expectedIds, RestHighLevelClient client, int numberOfShards)
+        throws IOException {
+        // restore index
+        RestoreSnapshotResponse restoreSnapshotResponse = client.snapshot()
+            .restore(
+                new RestoreSnapshotRequest("testrepo", "snap1").indices("test")
+                    .renamePattern("(.+)")
+                    .renameReplacement("restored_$1")
+                    .waitForCompletion(true),
+                RequestOptions.DEFAULT
+            );
+        assertNotNull(restoreSnapshotResponse.getRestoreInfo());
+        assertEquals(numberOfShards, restoreSnapshotResponse.getRestoreInfo().totalShards());
+        assertEquals(numberOfShards, restoreSnapshotResponse.getRestoreInfo().successfulShards());
+
+        // run a search against the index
+        assertDocs("restored_test", numDocs, expectedIds, client);
+
+        // mount as searchable snapshot
+        RestoreSnapshotResponse mountSnapshotResponse = client.searchableSnapshots()
+            .mountSnapshot(
+                new MountSnapshotRequest("testrepo", "snap1", "test").storage(MountSnapshotRequest.Storage.FULL_COPY)
```

Comment:
Can we randomize this between full/partial?

Reply:
needed another small tweak. I have this already in the follow-up.